src/libsyntax/parse/lexer/mod.rs

```diff
@@ -460,18 +454,6 @@ impl<'a> StringReader<'a> {
             let next_ch = char_at(&self.src, next_src_index);
             let next_ch_len = next_ch.len_utf8();
 
-            if self.ch.unwrap() == '\n' {
-                if self.save_new_lines_and_multibyte {
-                    self.filemap.next_line(self.next_pos);
```

Comment:
This checks if the current character is a newline, but then it passes `next_pos` into `next_line`